### DISCUSSION TITLE
quickFind 字段无法查询的条件

### DISCUSSION THREAD

**Discussion by aoguai** [gkd-kit/gkd#526] [General]
从 https://github.com/gkd-kit/docs/blob/99260a76265dd2f6a9f000917b9ff5e401a20418/api/src/index.ts#L190-L214 的描述来看
quickFind 字段在某些场景下即使存在符合要求的元素，也可能出现无法找到的情况。
这导致我很长时间内不太有百分百的信心敢放心使用该字段。
出于此原因我尝试搜索在什么场景下`findAccessibilityNodeInfosByViewId`和`findAccessibilityNodeInfosByText`会出现失效的情况。

最终我发现只要是在 **webview 内的控件**都无法通过`findAccessibilityNodeInfosByViewId`和`findAccessibilityNodeInfosByText`查找到。
具体可以看[这篇](https://zhuanlan.zhihu.com/p/678594793)知乎解释

我简单写了一个测试APP来测试，发现确实如此，下面是快照
- https://i.gkd.li/import/14318499

使用如下规则可以正常点击 webview 内的控件，而添加上 quickFind 字段后将失效。
```
{
  id: 'com.example.test',
  name: 'test',
  groups: [
    {
      key: 1,
      name: '[ChangeMe]规则名称-2024-02-17 18:33:28',
      desc: '[ChangeMe]本规则由GKD网页端审查工具生成',
      // quickFind: true,
      rules: [
        {
          activityIds: 'com.example.test.MainActivity',
          matches: '[text="后端"]',
          exampleUrls: 'https://m.gkd.li/34203474/11baf5a0-4c72-4971-b3da-0dca9ec423a6',
          snapshotUrls: 'https://i.gkd.li/import/14318499',
        },
      ],
    },
  ],
}
```

----

同时在已知规则中存在大量在对 webview 内的控件 添加 quickFind 字段 的情况，实际上都是无法查询到的。

- 虽然现在您已经不再维护规则，但是**我觉得可以在文档或字段说明中说明这一点，提醒其他开发者**

- 可以考虑优化一下快照逻辑和 inspect 审查工具对应的判断逻辑，我发现在许多旧快照中即使一些元素可以快速查找 inspect 审查工具也不一定显示提示，而是在 https://github.com/gkd-kit/inspect/commit/f23ac7bb1a4da50f99351e6be5acc5e15ca8a58b 这次更新后的快照才支持显示

**Comment by lisonge** [comment]
> 只要是在 webview 内的控件都无法通过

基于 compose-ui/flutter/以及一些sdk 内部都不支持，webview 并不是判断的唯一条件

**Reply by aoguai** [reply]
> 基于 compose-ui/flutter/以及一些sdk 内部都不支持，webview 并不是判断的唯一条件

嗯嗯，我的意思是目前来看 编写规则的话从 webview 是一个很容易判断的条件。但是不是唯一条件。

**Comment by AIsouler** [comment]
那像这种是不是不能使用qiuckfind 
https://i.gkd.li/i/14560546
[text="反馈"] + @View[visibleToUser=true] > Image <<n [id="android:id/content"]

**Reply by aoguai** [reply]
应该是

**Reply by lisonge** [reply]
`<< n` 就是遍历顺序，你可以理解为 `TextView` 完全等价于 `TextView <<n [depth=0]`

`android:id/content` 是每个应用的界面基本都会存在的靠近根部的节点，和 `[depth=0]` 没什么区别

**Reply by AIsouler** [reply]
明白了，写和不写没区别
